--- /tmp/trio-1.14/FILES	2009-10-31 16:51:57.000000000 +0100
+++ ./FILES	2013-03-15 20:23:02.000000000 +0100
@@ -2,7 +2,6 @@
 README
 CHANGES
 Makefile.in
-maketgz
 strio.h
 trio.c
 trio.h
@@ -13,21 +12,8 @@
 triostr.c
 triostr.h
 compare.c
-example.c
 regression.c
 configure
 configure.in
 install-sh
 autogen.sh
-doc/doc.h
-doc/doc_dynamic.h
-doc/doc_printf.h
-doc/doc_register.h
-doc/doc_scanf.h
-doc/doc_static.h
-doc/footer.html
-doc/header.html
-doc/trio.cfg
-doc/trio.css
-html/trio.css
-html/*.html
--- /tmp/trio-1.14/triop.h	2009-10-31 16:51:57.000000000 +0100
+++ ./triop.h	2013-03-15 20:12:57.000000000 +0100
@@ -299,11 +299,11 @@
  * separately.
  */
 #if defined(TRIO_MINIMAL)
-# if !defined(TRIO_EMBED_NAN)
-#  define TRIO_EMBED_NAN
+# if defined(TRIO_EMBED_NAN)
+#  undef TRIO_EMBED_NAN
 # endif
-# if !defined(TRIO_EMBED_STRING)
-#  define TRIO_EMBED_STRING
+# if defined(TRIO_EMBED_STRING)
+#  undef TRIO_EMBED_STRING
 # endif
 #endif
   
